### PR TITLE
Segments action validate user input

### LINKED ISSUE #522: Request input binding and validation
We'll need to add user input validation on all of our action endpoints in a standardized way.

Use gin's model binding and validation as described here: https://github.com/gin-gonic/gin#model-binding-and-validation

### PR DESCRIPTION
**Pull request by Dzalevski**
# Pull Request Template

## Description

Segments user input validation

Fixes #522 

## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test A Insomnia Calls
- [ ] Test B

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: entities/params/segmentsubs.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package params
+
+type SegmentSubs struct {
+	Ids []int64 `form:"ids[]" validate:"required"`
```
Comment (FilipNikolovski) [inline entities/params/segmentsubs.go:4]: @Dzalevski check out the `dive` parameter from validator: https://pkg.go.dev/github.com/go-playground/validator/v10#hdr-Dive

This allows us to dive into the slice and apply a validator on each element.

File: entities/params/segmentparams.go [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package params
+
+import (
+	"strings"
+)
+
+// SegmentParams represents request body for POST /api/segments
+type SegmentParams struct {
```
Comment (FilipNikolovski) [inline entities/params/segmentparams.go:8]: We'll need to agree on some naming here, on some params we prefix them with the http method like PostFoo, PutBar etc. and here it's just SegmentParams.

I think it's better if we prefix them with the HTTP method.

**Review by FilipNikolovski — CHANGES_REQUESTED** [review]

File: entities/params/segmentparams.go (hunk 2 above)
Comment (Dzalevski) [reply entities/params/segmentparams.go:8]: @FilipNikolovski  Yes this is the same case as before Segment is using same params for put,post

Comment (Dzalevski) [reply entities/params/segmentparams.go:8]: @FilipNikolovski In this case i think it's better to have 1 struct

**Review by FilipNikolovski — APPROVED** [review]

**Pull request merged**